File: 이상한문자만들기/java/20221020/src/Solution.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

class Solution {
    public String solution(String s) {
        String answer = process(s);
        return answer;
    }

    public String process(String string) {
        Pattern pattern = Pattern.compile("\\w+");
        Matcher matcher = pattern.matcher(string.toUpperCase());

        return matcher.replaceAll(x -> lowerCaseAtOddIndex(x.group()));
    }

    public String lowerCaseAtOddIndex(String string) {
        return IntStream.range(0, string.length())
                .mapToObj(i -> i % 2 == 0
                        ? string.substring(i, i + 1)
                        : string.substring(i, i + 1).toLowerCase())
                .collect(Collectors.joining());
    }
}

File: 이상한문자만들기/javascript/20221020/src/solution.js
function solution(s) {
  const answer = changeCase(s);
  return answer;
}

export function lowerCaseAtOddIndex(text) {
  return text.split('')
    .map((character, index) => (
      index % 2 === 0 ? character : character.toLowerCase()
    )).join('');
}

export function changeCase(string) {
  return string.toUpperCase().replace(/\w+/g, lowerCaseAtOddIndex);
}
